src/main/java/com/contrast/labs/ai/mcp/contrast/AssessService.java

```diff
@@ -889,31 +889,92 @@ private List<Metadata> getMetadataFromApp(Application app) {
   }
 
   /**
-   * Fetches all traces from the SDK by iterating through all pages. This is needed when in-memory
-   * session filtering is required, because we must have the complete dataset before filtering.
+   * Maximum traces to fetch for session filtering to prevent memory exhaustion. Configurable via
+   * property for testing. Default: 50,000.
+   */
+  @Value("${contrast.max-traces-for-session-filtering:50000}")
+  private int maxTracesForSessionFiltering = 50_000;
```

Comment:
usually class level variables that spring inject the value into are defined at the top of the class